modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/transaction/Branch.java

```diff
@@ -51,16 +49,34 @@
         SUSPENDED,
```

Comment:
This look like a state of the session rather than a state of the branch. Maybe we should separate out those two state sets.

Reply:
Just sharing what I came up with

#### Branch states
```
@startuml
[*] --> Active
Active --> Prepare
Active -> Rollback_Only
Prepare -> Prepared
Rollback_Only -> Forgotten
Prepared -> Forgotten
Forgotten --> [*]
@enduml
```

![branch-states](https://user-images.githubusercontent.com/1148405/37191824-fa48c8dc-2387-11e8-8952-e3e97c9db640.png)

#### Session states

```
@startuml
[*] --> Active
Active --> Suspend
Suspend : end(suspend)
Active : start(join)
Active : start()
Suspend --> [*]
@enduml
```
![session-states](https://user-images.githubusercontent.com/1148405/37192127-b8dd87aa-2389-11e8-8ed4-3a5d49ab236a.png)